code/obj/item/tank.dm

```diff
@@ -140,6 +203,26 @@ Contains:
 	process()
 		//Allow for reactions
 		if (air_contents)
+			// Exchange with the open air if the flap is open
+			if (src.get_tank_open())
+				// Ignore if unsimmed
+				var/turf/simulated/T = get_turf(src)
+				if (issimulatedturf(T))
+					// Borrows from transfer valve code
+					var/datum/gas_mixture/temp
+
+					if (!isnull(T) && !isnull(src.air_contents))
+						// Remove *some* of the gas from the tank at a time to 'leak' out
+						temp = src.air_contents.remove_ratio(0.25)
+						// Add the gas removed from the tank *all* into the turf
+						T.air.merge(temp)
+						// Determine what percentage of the turf's air contents should go back into the tank
+						var/transfer_ratio = src.air_contents.volume / (src.air_contents.volume + T.air.volume)
+						// Remove aformentioned amount from the turf, put it back into the temp
+						temp = T.air.remove_ratio(transfer_ratio)
+						// Lastly, put all that air removed from the combined mixture back into the tank.
+						src.air_contents.merge(temp)
```

Comment:
Does this work? I think you'll need to involve a pressure delta somewhere or the tank won't equalise to ambient pressure.

Tried calculating to check but IDK what the default contents of a tile is in moles.

Reply:
ah yeah it borks up a bit. gimme a minute

Reply:
![image](https://github.com/user-attachments/assets/ad82f49f-54e6-4171-aeb8-6643b558a22d)
Probably going to do something like this instead